modules/core/02-client/types/exp/hooks.go

```diff
@@ -0,0 +1,56 @@
+package exp
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/ibc-go/v7/modules/core/exported"
+)
+
+// Hooks specifies methods to enable writing custom logic which executes
+// after specific client lifecycle methods.
+// Note: this interface is experimental and could be removed in favour of a different solution, requiring code changes.
+type Hooks interface {
+	// OnClientCreated is executed upon client creation.
+	OnClientCreated(ctx sdk.Context, clientID string) error
+	// OnClientUpdated is executed when a client is updated.
+	OnClientUpdated(ctx sdk.Context, clientID string, consensusHeights []exported.Height) error
```

Comment:
```suggestion
	OnClientUpdated(ctx sdk.Context, clientID string, clientMsg exported.ClientMessage, consensusHeights []exported.Height) error
```

Reply:
This is so the hook also has access to the header which in the tendermint case has the unhashed validator set as well as other potentially useful information like `DataHash`